Firebase.cpp

```diff
@@ -15,100 +15,158 @@
 //
 #include "Firebase.h"
 
-const char* firebaseFingerprint = "7A 54 06 9B DC 7A 25 B3 86 8D 66 53 48 2C 0B 96 42 C7 B3 0A";
-const uint16_t firebasePort = 443;
+namespace {
+const char* kFirebaseFingerprint = "7A 54 06 9B DC 7A 25 B3 86 8D 66 53 48 2C 0B 96 42 C7 B3 0A";
+const uint16_t kFirebasePort = 443;
+}  // namespace
 
-Firebase::Firebase(const String& host) : _host(host) {
-  _http.setReuse(true);
+Firebase::Firebase(const String& host) : connection_(host) {
 }
 
 Firebase& Firebase::auth(const String& auth) {
-  _auth = auth;
+  connection_.auth(auth);
   return *this;
 }
 
-String Firebase::get(const String& path) {
-  return sendRequest("GET", path);
+FirebaseResult Firebase::get(const String& path) {
```

Comment:
> the problem is this style assumes the calls succeed, which isn't great for remote services.

I tried to add explicit error checking after each call in every sample, but I agree with the sentiment.

An intermediate result object make things more discoverable (they have to lookup the doc for the result, and then it's easier for them to find the error methods here, rather than when it's hidden inside the main class), but they still need to remember to call `isError()` on it, and a good way to "learn" that pattern is to repeat it across sample.